File: programmers/02.stack,queue/01.같은 숫자는 싫어 /python/other.py
def solution(arr):
    a = []
    for i in arr:
        if a[-1:] == [i]: 
            continue
        a.append(i)
    return a

# test
print(solution([1,1,3,3,0,1,1]))
print(solution([4,4,4,3,3]))

File: programmers/01.알고리즘/02.stack,queue/02.올바른 괄호/JS/other.js
function solution(s) {
    let cum = 0;
    for (let paren of s) {
        console.log(paren);
        cum += paren === "(" ? 1 : -1;
        if (cum < 0) {
            return false;
        }
    }
    return cum === 0 ? true : false;
}

// test
console.log(solution("()()"));
console.log(solution("(())()"));
console.log(solution(")()("));
console.log(solution("(()("));
